crates/containerd-shim-wasm/src/sandbox/shim.rs

```diff
@@ -1207,6 +1214,59 @@ where
         }
         Ok(state)
     }
+
+    fn task_stats(&self, req: StatsRequest) -> Result<StatsResponse> {
+        let i = self.get_instance(req.id())?;
+        let pid_lock = i.pid.read().unwrap();
+        let pid = *pid_lock;
+        if pid.is_none() {
+            return Err(Error::InvalidArgument("task is not running".to_string()));
+        }
+
+        let mut metrics = Metrics::new();
+        let hier = hierarchies::auto();
+        let cgroup: Cgroup;
+        if hier.v2() {
```

Comment:
Fun  fact, `hier` means "yesterday" in french :)

Reply:
😮